src/test/java/teammates/test/cases/datatransfer/FeedbackResponseAttributesTest.java

```diff
@@ -44,4 +48,126 @@ public void testDefaultTimestamp() {
+                .withFeedbackSessionName(expectedSessionName)
+                .withCourseId(expectedCourseId)
+                .withFeedbackQuestionId(expectedQuestionId)
+                .withFeedbackQuestionType(expectedQuestionType)
+                .withGiver(expectedGiver)
+                .withGiverSection(expectedGiverSection)
+                .withRecipient(expectedRecipient)
+                .withRecipientSection(expectedRecipientSection)
+                .withResponseMetaData(expectedResponseMetaData)
+                .withCreatedAt(expectedCreatedAt)
+                .withUpdatedAt(expectedUpdatedAt)
+                .build();
+
+        assertEquals(expectedId, observedFeedbackResponseAttributes.getId());
+        assertEquals(expectedSessionName, observedFeedbackResponseAttributes.feedbackSessionName);
+        assertEquals(expectedCourseId, observedFeedbackResponseAttributes.courseId);
+        assertEquals(expectedQuestionId, observedFeedbackResponseAttributes.feedbackQuestionId);
+        assertEquals(expectedQuestionType, observedFeedbackResponseAttributes.feedbackQuestionType);
+        assertEquals(expectedGiver, observedFeedbackResponseAttributes.giver);
+        assertEquals(expectedRecipient, observedFeedbackResponseAttributes.recipient);
+        assertEquals(expectedGiverSection, observedFeedbackResponseAttributes.giverSection);
+        assertEquals(expectedRecipientSection, observedFeedbackResponseAttributes.recipientSection);
+        assertEquals(expectedResponseMetaData, observedFeedbackResponseAttributes.responseMetaData);
+        assertEquals(expectedCreatedAt, observedFeedbackResponseAttributes.getCreatedAt());
+        assertEquals(expectedUpdatedAt, observedFeedbackResponseAttributes.getUpdatedAt());
+    }
+
+    @Test
+    public void testBuilderCopy() {
+        String originalGiver = "giver";
+        String originalRecipient = "recipient";
+        String originalQuestionId = "someQuestionId";
+        String originalId = originalQuestionId + "%" + originalGiver + "%" + originalRecipient;
+        FeedbackResponseAttributes original = FeedbackResponseAttributes.builder()
+                .withFeedbackResponseId(originalId)
+                .withFeedbackResponseId("originalFeedbackQuestionId%originalGiver%")
+                .withFeedbackSessionName("originalName")
+                .withCourseId("originalCourseId")
+                .withFeedbackQuestionId(originalQuestionId)
+                .withFeedbackQuestionType(FeedbackQuestionType.TEXT)
+                .withGiver(originalGiver)
+                .withRecipient(originalRecipient)
+                .withResponseMetaData(new Text("original meta data"))
+                .withCreatedAt(Instant.now())
+                .withUpdatedAt(Instant.now())
+                .build();
+
+        FeedbackResponseAttributes copy = original.getCopy();
+
+        assertNotSame(copy, original);
+        assertEquals(copy.getId(), original.getId());
```

Comment:
Do an `assertSame` for every field and mention that copying of immutable fields only entails copying of references.

Reply:
`assertSame` doesn't accept `String` as arguments?

Reply:
A `String` is still an `Object` but we don't have it in `BaseTestCase`, we should probably get rid and all of them and change to importing them in the future.

Reply:
I agree with @LiHaoTan actually. With IDEs importing isn't much trouble. It's more troublesome to have to add code to `BaseTestCase` every time we want to use a previously-unused method found in `org.junit.Assert` (and there are so many of them). If importing is troublesome, we can always allow
```java
import static org.junit.Assert.*;
```
for test classes.

Reply:
I don't get it yet. How do I add `String` to `BaseTestCase`?

Reply:
I mean add:
```Java
protected static void assertSame(Object expected, Object actual) {
    AssertJUnit.assertSame(expected, actual);
}
```

Reply:
> With IDEs importing isn't much trouble.

@whipermr5 there goes your first (wrong) assumption: _with IDEs_. I know in this day and age IDEs are common, but unless we prescribe that one "must use an IDE" to work in this project, this condition has no guarantee.

> It's more troublesome to have to add code to `BaseTestCase` every time we want to use a previously-unused method found in `org.junit.Assert` (and there are so many of them).

Is the probability of one finding a method not copied in `BaseTestCase` really higher than the probability that one needs to import something from `org.junit.Assert` if we go down the other path? Besides, if this is really a problem, the solution is simple: copy all the methods of `Assert` class.

Reply:
> I know in this day and age IDEs are common, but unless we prescribe that one "must use an IDE" to work in this project, this condition has no guarantee.

I know we don't make IDEs a must. Here, it's still possible to import everything manually without an IDE, just as how it's possible to rename a variable across multiple files manually without an IDE. I feel we shouldn't make decisions based on how troublesome something is without an IDE, as long as it's still possible to do without an IDE. In choosing not to use an IDE, the developer would have to accept the additional trouble. (If we'd like to make life easier for them, we can always allow the static wildcard import for test files, as previously mentioned.)

> Besides, if this is really a problem, the solution is simple: copy all the methods of Assert class.

There's some effort needed to keep the methods in sync. Also, if the javadoc for each method is not copied over, we lose easy access to it; if it is, it's an additional thing to keep in sync. An example of consequences when things are not exactly in sync: while working on #8759, I was really confused when I wanted to use this method because of the parameter names (what does "expected" and "actual" mean when we're asserting that the two things are *not* equal?):
https://github.com/TEAMMATES/teammates/blob/25008d19fd7c81a78780112996617e1f5abc4628/src/test/java/teammates/test/cases/BaseTestCase.java#L178-L180
After a while I realised that the confusing parameter names were a mistake when copying `Assert#assertNotEquals` - the original method's parameter names were, much more understandably, "first" and "second".

Why bother keeping them in sync when we can just import them? Isn't this the whole point of the import system? I don't think importing should be seen as something painful. Also I'm pretty sure the framework designers intended for these methods to be imported exactly where they are used.

Finally, this may not be completely related, but after reading it in recent times I'm more wary about using inheritance in test code: https://www.petrikainulainen.net/programming/unit-testing/3-reasons-why-we-should-not-use-inheritance-in-our-tests/

Reply:
Okay, I just read your original issue #5362 for context; I must admit those are some pretty strong points too.